aklogger/__init__.py

```diff
@@ -1,57 +1,150 @@
 import logging
-from . import formatters, handlers, utils
+
+from slacker import Slacker
+
+tpl = """
+=====================
+{summary}
+---------------------
+{details}
+=====================
+"""
+
+
+def getBaseLogger(name):
+    return logging.getLogger(name)
+
+
+def getBaseCeleryLogger(name):
+    return get_task_logger(name)
+
+
+get_logger = getBaseLogger
+
+try:
+    from celery.utils.log import get_task_logger
+    get_logger = getBaseCeleryLogger
+except ImportError:
+    pass
+
+CRITICAL = logging.CRITICAL
+ERROR = logging.ERROR
+WARNING = logging.WARNING
+INFO = logging.INFO
+DEBUG = logging.DEBUG
+NOTSET = logging.NOTSET
+
+_levelToName = {
+    CRITICAL: 'CRITICAL',
+    ERROR: 'ERROR',
+    WARNING: 'WARNING',
+    INFO: 'INFO',
+    DEBUG: 'DEBUG',
+    NOTSET: 'NOTSET',
+}
 
 
 class AKLogger(object):
 
-    def __init__(self, name='root', level=None, **kwargs):
-        self.logger = logging.Logger(name)
-        self.handler = kwargs.get('handler', handlers.StreamHandler())
-        self.formatter = kwargs.get('formatter', formatters.basic_formatter)
-        if not level:
-            self.level = 'DEBUG'
-        self.handler.setLevel(self.level)
-        self.handler.setFormatter(self.formatter)
-        self.logger.addHandler(self.handler)
-
-    def set_level(self, level):
-        return self.logger.setLevel(level)
-
-    def debug(self, title=None, msg=None, *args, **kwargs):
-        message = utils.get_message(title, msg)
-        if not message:
-            return
-        return self.logger.debug(message)
+    def __init__(self, name, parent=None):
+        self.set_name(name)
+        self.parent = parent
+        self.logger = get_logger(self.get_name())
```

Comment:
@cosmosgenius, there is bit of problem here, its always importing the celery logger when there is a celery installed(which will always be the case in mycroft). So in that case our normal logger is not logging to console until it's a celery task.

P.S. file log and slack push is working in both the cases.